src/audio/kpb.c

```diff
@@ -1648,11 +1707,15 @@ static void kpb_buffer_samples(const struct audio_stream __sparse_cache *source,
 		samples_count = KPB_BYTES_TO_S16_SAMPLES(size);
 		samples_offset = KPB_BYTES_TO_S16_SAMPLES(offset);
 		audio_stream_copy_to_linear(source, samples_offset,
-					    sink, 0, samples_count);
+				    sink, 0, samples_count);
 		break;
 #endif
 #if CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE
 	case 24:
+		samples_count = size / ((sample_width / 8) * source->channels);
+		samples_offset = offset / ((sample_width / 8) * source->channels);
+		kpb_convert_32b_to_24b(source, samples_offset,
+				       sink, 0, samples_count);
```

Comment:
Consider adding _fallthrough_ annotation.

Reply:
isn't a `break` missing here in fact? Looks very much like it is.